runtime/polkadot/src/lib.rs

```diff
@@ -1296,20 +1300,47 @@ impl parachains_paras::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type WeightInfo = weights::runtime_parachains_paras::WeightInfo<Runtime>;
 	type UnsignedPriority = ParasUnsignedPriority;
+	type UmpQueueTracker = ParaInclusion;
 	type NextSessionRotation = Babe;
 }
 
 parameter_types! {
-	pub const FirstMessageFactorPercent: u64 = 100;
+	/// Amount of weight that can be spent per block to service messages.
+	/// FAIL-CI: Pretty random value.
+	pub const MessageQueueServiceWeight: Weight = Weight::from_parts(100 * WEIGHT_REF_TIME_PER_MILLIS, u64::MAX);
+}
+
+pub struct MessageProcessor;
+impl ProcessMessage for MessageProcessor {
+	type Origin = AggregateMessageOrigin;
+
+	fn process_message(
+		message: &[u8],
+		origin: Self::Origin,
+		weight_limit: Weight,
+	) -> Result<(bool, Weight), ProcessMessageError> {
+		let para = match origin {
+			AggregateMessageOrigin::UMP(para) => para,
+		};
+		xcm_builder::ProcessXcmMessage::<
+			Junction,
+			xcm_executor::XcmExecutor<xcm_config::XcmConfig>,
+			RuntimeCall,
+			// FAIL-CI why is the `into` needed?
+		>::process_message(message, Junction::Parachain(para.into()), weight_limit)
```

Comment:
Not sure here with the Junction. Why does `Parachain` take a `u32` and not a `ParaId`?  
This impl is also copy&pasted into each runtime.

Reply:
Because the `Junction` is defined in XCM which is upstream of Polkadot, Cumulus and its core types.